src/riak_logger_config.erl

```diff
@@ -0,0 +1,471 @@
+%% -------------------------------------------------------------------
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+%% Portions informed by OTP module logger_filters,
+%% Copyright Ericsson AB 2017-2018, though no code is duplicated.
+%%
+%% @doc Riak kernel logger configuration functions
+%% 
+%% These functions should be two arity, and pass some form of configuration
+%% map, and a function for extracting the values of string-based keys from
+%% that map (e.g. cuttlefish:conf_get/2).
+%% 
+%% The configuration can be ignored if preferred - and a statically defined
+%% logging config be returned
+
+-module(riak_logger_config).
```

Comment:
Maybe refactor into more descriptive module name(s), to allow for additions without namespace confusion? Possible names:
`riak_log_std_config`
`riak_log_type_config`
`riak_log_json_config`
That suggests there may be a shared set of functions, perhaps in `riak_log_common_config`.